backend/internal/storage/database.go

```diff
@@ -0,0 +1,55 @@
+package storage
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+// Configuration to connect to database
+type DsnConfig struct {
+	// Username to login database
+	User string
+	// User password to verify user
+	Password string
+	// Database name to be accessed
+	DatabaseName string
+	// The host used to host database
+	Host string
+	// The port used to connect database
+	Port uint16
+}
+
+// Create a valid DSN string to connect database from DsnConfig
+func (config *DsnConfig) NewDsn() string {
+	dsn := fmt.Sprintf(
+		"user=%s password=%s dbname=%s host=%s port=%d",
+		config.User, config.Password, config.DatabaseName, config.Host, config.Port,
+	)
+	return dsn
+}
```

Comment:
Now that you made this into a method, I think better name for this could be just `String()` as it only converts it into a string form.

Reply:
That was kind of the idea since `DsnConfig` only job at moment is to be a typed interface for the DSN string.